backend/src/main/java/com/benine/backend/http/EditPresetHandler.java

```diff
@@ -71,34 +81,78 @@ public void handle(String s, Request request, HttpServletRequest req, HttpServle
    * Updating the tag only.
    * @param preset the preset to be changed
    * @param tagList the tag to be added
+   * @return the tag added to the preset
    */
-  public void updateTag(Preset preset, List<String> tagList) {
+  public Set<String> updateTag(Preset preset, List<String> tagList) {
     preset.removeTags();
     preset.addTags(tagList);
+    return preset.getTags();
   }
   
   /**
    * Editing an already existing preset by removing the old preset and creating a new 
    * preset with the same preset and camera id. It also creates a new image that belongs to the 
    * preset and updates the database.
-   * @param camID                         The id of the camera object. 
-   * @param tagList                       The tags belonging to the preset.
    * @param presetID                      The id of the preset. 
+   * @param preset                        The preset to be updated.
    * @throws IOException                  If the image cannot be created.
    * @throws StreamNotAvailableException  If the camera does not have a stream.
    * @throws SQLException                 If the preset cannot be written to the database.
    * @throws CameraConnectionException    If the camera cannot be reached.
    * @throws MalformedURIException        If there is an error in the request.
    */
-  public void updatePosition(String camID, List<String> tagList, int presetID) throws 
+  public void updatePosition(Preset preset, int presetID) throws 
```

Comment:
This method can be private, and most of the other methods in this class as well.